cloudevents/sdk/event/base.py

```diff
@@ -128,13 +141,16 @@ def UnmarshalBinary(
         body: typing.IO,
         data_unmarshaller: typing.Callable
     ):
+        keys = set(headers)
```

Comment:
Can we have a more descriptive name than `keys`?

Like `unique_headers`. Not sure why we are doing this.